src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishTrimmedWindowsFormsAndWPFApps.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using FluentAssertions;
+using Microsoft.NET.Build.Tasks;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.Publish.Tests
+{
+    public class GivenThatWeWantToPublishTrimmedWindowsFormsAndWPFApps : SdkTest
+    {
+        public GivenThatWeWantToPublishTrimmedWindowsFormsAndWPFApps(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        [WindowsOnlyRequiresMSBuildVersionFact("16.8.0")]
+        public void It_publishes_and_err_windows_forms_app()
+        {
+            var targetFramework = "net6.0-windows";
+            TestProject testProject = new TestProject()
+            {
+                Name = "WinformsFailTest",
+                ProjectSdk = "Microsoft.NET.Sdk", 
+                TargetFrameworks = targetFramework
+            };
+            testProject.AdditionalProperties["UseWindowsForms"] = "true";
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+            
+            var publishCommand = new PublishCommand(testAsset);
+            publishCommand.Execute($"/p:PublishTrimmed=true")
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining(Strings.@TrimmingWindowsFormsRequiresCom);
```

Comment:
We usually assert that the error code is present instead of the string itself (for example: `.HaveStdOutContaining("NETSDK1165")`